test/prompt/invocation_layer/test_sagemaker_hf_infer.py

```diff
@@ -0,0 +1,409 @@
+import os
+from unittest.mock import patch, MagicMock, Mock
+
+import pytest
+
+from haystack.lazy_imports import LazyImport
+
+from haystack.errors import SageMakerConfigurationError
+from haystack.nodes.prompt.invocation_layer import SageMakerHFInferenceInvocationLayer
+
+with LazyImport() as boto3_import:
+    from botocore.exceptions import BotoCoreError
+
+
+# create a fixture with mocked boto3 client and session
+@pytest.fixture
+def mock_boto3_session():
+    with patch("boto3.Session") as mock_client:
+        yield mock_client
+
+
+@pytest.fixture
+def mock_prompt_handler():
+    with patch("haystack.nodes.prompt.invocation_layer.handlers.DefaultPromptHandler") as mock_prompt_handler:
+        yield mock_prompt_handler
+
+
+@pytest.mark.unit
+def test_default_constructor(mock_auto_tokenizer, mock_boto3_session):
+    """
+    Test that the default constructor sets the correct values
+    """
+
+    layer = SageMakerHFInferenceInvocationLayer(
+        model_name_or_path="some_fake_model",
+        max_length=99,
+        aws_access_key_id="some_fake_id",
+        aws_secret_access_key="some_fake_key",
+        aws_session_token="some_fake_token",
+        aws_profile_name="some_fake_profile",
+        aws_region_name="fake_region",
+    )
+
+    assert layer.max_length == 99
+    assert layer.model_name_or_path == "some_fake_model"
+
+    # assert mocked boto3 client called exactly once
+    mock_boto3_session.assert_called_once()
+
+    # assert mocked boto3 client was called with the correct parameters
+    mock_boto3_session.assert_called_with(
+        aws_access_key_id="some_fake_id",
+        aws_secret_access_key="some_fake_key",
+        aws_session_token="some_fake_token",
+        profile_name="some_fake_profile",
+        region_name="fake_region",
+    )
+
+
+@pytest.mark.unit
+def test_constructor_with_model_kwargs(mock_auto_tokenizer, mock_boto3_session):
+    """
+    Test that model_kwargs are correctly set in the constructor
+    and that model_kwargs_rejected are correctly filtered out
+    """
+    model_kwargs = {"temperature": 0.7, "do_sample": True, "stream": True}
+    model_kwargs_rejected = {"fake_param": 0.7, "another_fake_param": 1}
+
+    layer = SageMakerHFInferenceInvocationLayer(
+        model_name_or_path="some_fake_model", **model_kwargs, **model_kwargs_rejected
+    )
+    assert "temperature" in layer.model_input_kwargs
```

Comment:
```suggestion
    assert layer.model_input_kwargs["temperature"] == 0.7
```
(untested)
It might be a more comprehensive check.